AppInspector/Commands/AnalyzeCommand.cs

```diff
@@ -335,6 +335,17 @@ void VerifyFile(string filename)
             }
         }
 
+        // Remove match records that don't have the matching depends on tags
+        List<MatchRecord> filteredMatchRecords = _metaDataHelper.Matches.Where(x => x.Rule?.DependsOnTags?.All(tag => _metaDataHelper.UniqueTags.ContainsKey(tag)) ?? true).ToList();
```

Comment:
@jacobmsft This particular check does not depend on the ID, it just cares about the tags - which are always allowed to be set by multiple rules.

Reply:
ah ok...I thought we were iterating on the rules, but its x.Rule., which makes sense.

Reply:
I did discover an edge case, but its actually from the `Overrides` not the `ID`.

Overrides are removed on a per file basis as each file is scanned.

DependsOnTags must be checked after all files are scanned, because it cares about tags in other files.

Lets say Rule A matches and does not have a DependsOnTags. Rule B also matches and overrides Rule A, however it has a DependsOnTags for "TagC".

At the end of the run, if TagC isn't present, Rule B will be removed from the matches set, but Rule A will not currently be restored. We do not keep track of the rules which were overridden and when, so this information is not available at the time we check `DependsOnTags` to remove rules with missing dependent tags.

The easiest way to fix this might be in the verifier, to require that a rule which is targetted with an override must depend on at least all of the tags which the overrider depends on.